src/coldfront_plugin_api/management/commands/migrate_fos.py

```diff
@@ -0,0 +1,79 @@
+from django.core.management.base import BaseCommand
+from coldfront.core.project.models import Project
+from coldfront.core.field_of_science.models import FieldOfScience
+
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = """Migrates Coldfront's list of fields of sciences (fos), changing what
+    fos a project can be assigned to, and updating the fos in all existing projects.
+
+    Requires a csv, tab-seperated, containing two columns, the first containing the list of old fos,
+    the second containing the new fos that the old fos will map onto.
+
+    I.e to map 'Quantum Mechanics' and 'Photonics' to 'Physics', provide this csv:
+        Quantum Mechanics   Physics
+        Photonics   Physics
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "-m",
+            "--mapping",
+            required=True,
+            help="required tab-seperated csv file to provide mapping for migration",
+        )
+
+    def handle(self, *args, **options):
+        # Get csv file
+        mapping_csv = options["mapping"]
+
+        mapping_dict = dict()
+        new_fos_set = set()
+        with open(mapping_csv, "r") as f:
+            for line in f:
+                old_fos, new_fos = line.strip().split("\t")
+                mapping_dict[old_fos] = new_fos
+                new_fos_set.add(new_fos)
+
+        # Validate list of old fos
+        invalid_fos = False
+        for old_fos_name in list(mapping_dict.keys()):
+            if not FieldOfScience.objects.filter(description=old_fos_name):
+                invalid_fos = True
```

Comment:
Though, instead of exiting here, this is a case where you can log a warning and proceed with execution. If there doesn't exist a source field of science, then that means that there are no projects using it to be migrated. 

A strong reason for the above is that this makes the code idempotent. Meaning that executing the same command several times produces the same outcome. That wouldn't be the case if you exit since the execution would exit with error when it doesn't find a Field of Science that wasn't already migrated.